src/runner/settings_window.cpp

```diff
@@ -83,11 +83,8 @@ void send_json_config_to_module(const std::wstring& module_key, const std::wstri
 
 void dispatch_json_config_to_modules(const json::JsonObject& powertoys_configs)
 {
-    for (const auto& powertoy_element : powertoys_configs)
-    {
-        const auto element = powertoy_element.Value().Stringify();
-        send_json_config_to_module(powertoy_element.Key().c_str(), element.c_str());
-    }
+    std::wstring pt_module_name = powertoys_configs.Lookup(L"name").GetString().c_str();
+    send_json_config_to_module(pt_module_name, powertoys_configs.Stringify().c_str());
```

Comment:
could you please help me understand this change?

Reply:
I guess this is to avoid sending back the updated settings to the module, and just notify the module that it needs to reload the settings.
If we want to be able to do a smooth transition to the new settings we should, for the time being, keep the existing logic otherwise we have to update all the modules implementation in one PR.

Reply:
@yuyoyuppe @enricogior 

 I will explain the motivation for the change helow :):
The old settings send this JSON to the runner. This JSON contains the name of the PowerToy/module as a key,  I could not find a way to create class models for powertoys modules with names that contains spaces in them. So i noticed that this key was redundant i.e it is also included the "name" key of the object value itself.
  
```json
{
    "powertoys":{
        "File Explorer Preview":{
            "name": "File Explorer Preview",
            "properties": {
              "IDS_PREVPANE_SVG_BOOL_TOGGLE_CONTROLL": { "value": true },
              "PREVPANE_MD_BOOL_TOGGLE_CONTROLL_ID": { "value": true }
            },
            "version": "1.0"
          }
    }
}
```
So I decided to send settings in the following format:
```
{
    "powertoys":{
        "name": "File Explorer Preview",
        "properties": {
          "IDS_PREVPANE_SVG_BOOL_TOGGLE_CONTROLL": { "value": true },
          "PREVPANE_MD_BOOL_TOGGLE_CONTROLL_ID": { "value": true }
        },
        "version": "1.0"
      }
}

```

With the old format: The runner would have to loop through root keys in the "powertoys" key object value to get the name of the module as shown here:
![image](https://user-images.githubusercontent.com/58791731/77729703-8b4f8280-6fbc-11ea-8825-9342138b364c.png)

With the new format, I updated the runner to get the name of the module from the "powertoys" object value itself in the attribute/key "name":

![image](https://user-images.githubusercontent.com/58791731/77729755-b3d77c80-6fbc-11ea-93c4-d853aed3ab36.png)

Is this a breaking change?

Reply:
It's a breaking change, because the runner will crash here when using the old settings, and, as @enricogior stated, we want to keep both versions of the settings app for a single transitional release. 

If you already have a way to determine from the runner which version of the settings it's currently using, just do an `if`/`else` on that. Otherwise, let's introduce `#ifdef SETTINGS_V2`/`#else` as a quick workaround for this branch, so any breaking change goes under that `#ifdef`. 

When it's time to merge `dev/settingsV2` into `master`, there could be a separate task to turn those compile-time `#ifdef`S into runtime `if`S.

Reply:
I just had a conversation with Clint and Ryan and we will not be using the old settings. We will be using the new settings only. I will create a task to update the old settings JSON schema sent out to match the new one.

Reply:
We can fix this without breaking backwards compatibility.
The most important thing is to preserve the JSON persisted schema.
For the in-memory schema of the data that we pass from the settings to the runner, we can use names without space in it.